src/libraries/System.Runtime.InteropServices/gen/DllImportGenerator/DllImportStubContext.cs

```diff
@@ -123,18 +125,31 @@ public IEnumerable<ParameterSyntax> StubParameters
 
             ImmutableArray<AttributeListSyntax>.Builder additionalAttrs = ImmutableArray.CreateBuilder<AttributeListSyntax>();
 
-            // Define additional attributes for the stub definition.
+            if (env.TargetFramework != TargetFramework.Unknown)
+            {
+                // Define additional attributes for the stub definition.
+                additionalAttrs.Add(
+                    AttributeList(
+                        SingletonSeparatedList(
+                            Attribute(ParseName(TypeNames.System_CodeDom_Compiler_GeneratedCodeAttribute),
+                                AttributeArgumentList(
+                                    SeparatedList(
+                                        new[]
+                                        {
+                                            AttributeArgument(LiteralExpression(SyntaxKind.StringLiteralExpression, Literal("Microsoft.Interop.DllImportGenerator"))),
```

Comment:
Since we're already using the DllImportGenerator assembly name to get the version string, we can probably just use it for the name too.